lib/Service/Classification/PersistenceService.php

```diff
@@ -114,8 +116,9 @@ public function persist(Classifier $classifier, Persistable $estimator): void {
 		 */
 		$tmpPath = $this->tempManager->getTemporaryFile();
 		try {
-			$persister = new Filesystem($tmpPath);
-			$persister->save($estimator);
+			/* @psalm-suppress InvalidArgument **/
+			$model = new PersistentModel($estimator, new Filesystem($tmpPath));
```

Comment:
Is required because the constructor is annotated a bit weirdly.

## Relevant code
```php
    /**
     * @param \Rubix\ML\Learner $base
     * @param \Rubix\ML\Persisters\Persister $persister
     * @param \Rubix\ML\Serializers\Serializer|null $serializer
     * @throws \Rubix\ML\Exceptions\InvalidArgumentException
     */
    public function __construct(Learner $base, Persister $persister, ?Serializer $serializer = null)
    {
        if (!$base instanceof Persistable) {
            throw new InvalidArgumentException('Base Learner must'
                . ' implement the Persistable interface.');
        }
```

## Relevant documentation
https://docs.rubixml.com/2.0/persistent-model.html

![grafik](https://user-images.githubusercontent.com/1479486/215492558-bf7f15f3-9e65-43fa-9981-63cf95df95bf.png)

Reply:
Statically speaking it should be a Learner. The dynamic type should also be a Persistable. We can use a union type Learner|Persistable in our code.